### ISSUE TITLE
Permission.bluetoothScan.request() return PermissionStatus.denied

### ISSUE THREAD

**Issue by JDongKhan** [Baseflow/flutter-permission-handler#1077]
### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-permission-handler/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-permission-handler/blob/main/permission_handler/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [X] Android
- [ ] iOS
- [ ] Windows

### Steps to reproduce

1、run permission_handler_android/example project
2、tap Permission.bluetoothScan
3、log  
```
I/flutter (20067): {Permission.accessNotificationPolicy: PermissionStatus.granted}
I/flutter (20067): PermissionStatus.denied

```

### Expected results

show PermissionStatus.granted

### Actual results

show PermissionStatus.denied

### Code sample

 run permission_handler_android/example project

```dart
Future<void> requestPermission(Permission permission) async {
    print(permission);
    final status = await _permissionHandler.requestPermissions([permission]);

    setState(() {
      print(status);
      _permissionStatus = status[permission] ?? PermissionStatus.denied;
      print(_permissionStatus);
    });
  }
```

### Screenshots or video

https://github.com/Baseflow/flutter-permission-handler/assets/4929805/e3ffd57f-664c-4555-b8ea-5a207ead553e

![苏宁豆芽图片20230630141532394](https://github.com/Baseflow/flutter-permission-handler/assets/4929805/77ec830b-6ce3-4329-9f5d-f8de1203299c)

### Version

10.3.0

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
/Users/sn/fvm/default/bin/flutter doctor --verbose
[✓] Flutter (Channel stable, 3.10.5, on macOS 13.0 22A380 darwin-x64, locale zh-Hans-CN)
    • Flutter version 3.10.5 on channel stable at /Users/sn/fvm/versions/3.10.5
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 796c8ef792 (2 weeks ago), 2023-06-13 15:51:02 -0700
    • Engine revision 45f6e00911
    • Dart version 3.0.5
    • DevTools version 2.23.1
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[!] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/sn/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode_14_2.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] IntelliJ IDEA Community Edition (version 2019.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.79.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.66.0

[✓] Connected device (4 available)
    • HEY W09 (mobile)   • A3ALUN2726G10922          • android-arm64  • Android 12 (API 31)
    • sn的iPhone (mobile) • 00008101-001E518E21FA001E • ios            • iOS 16.2 20C65
    • macOS (desktop)    • macos                     • darwin-x64     • macOS 13.0 22A380 darwin-x64
    • Chrome (web)       • chrome                    • web-javascript • Google Chrome 110.0.5481.100

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
Process finished with exit code 0
```

</details>

**Comment by PNThanggg** [comment]
I also got the same bug.

**Comment by JeroenWeener** [comment]
Thanks for reporting this issue and taking the time to provide the necessary information @JDongKhan!

Unfortunately, I am unable to reproduce it. I am running `permission_handler` `10.3.0`'s example application on an Android 13 (API 33) emulator.

https://github.com/Baseflow/flutter-permission-handler/assets/9049645/42c92c6c-9088-40de-ac5a-28d75c4a2c80

What device (tablet?) are you running the application on?

**Comment by mvanbeusekom** [comment]
This might be related to a bug that was released in the `permission_handler_platform_interface` package (version 3.11.0) recently. We released a hotfix (version 3.11.0+1) shortly after, however if the application if using the wrong dependency this might still go wrong. 

Can you verify in the `pubspec.lock` file which version of the `permission_handler_platform_interface` package is being used? 

If this is not version `3.11.0+1` please run the `flutter pub upgrade` command to ensure the package is being updated to the latest version. 

We would really appreciate it if you could confirm.

**Comment by JDongKhan** [comment]
@mvanbeusekom  
Thank you，i have upgrade 3.11.0+1 and It's work for me.
